### PR TITLE
Task-39254: Display detected positive items on the quarantine page

### PR DESCRIPTION
**Pull request by AzmiTouil**

### CODE REVIEW

File: commons-dlp/pom.xml [hunk 1]
```diff
@@ -26,5 +26,11 @@
       <artifactId>commons-testing</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+        <groupId>org.exoplatform.social</groupId>
+        <artifactId>social-component-core</artifactId>
+        <version>6.2.x-ma-SNAPSHOT</version>
```
Comment (rdenarie) [inline commons-dlp/pom.xml]: you should not have a version here

Comment (rdenarie) [reply commons-dlp/pom.xml]: In addition, commons should not depends on social, as social depends on commons

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java [hunk 2]
```diff
@@ -21,8 +29,18 @@ public DlpPositiveItemServiceImpl(DlpPositiveItemDAO dlpPositiveItemDAO) {
 
 
     @Override
-    public List<DlpPositiveItemEntity> getDlpPositivesItems() {
-        return dlpPositiveItemDAO.findAll();
+    public DlpPositiveItemList getDlpPositivesItems(int offset, int limit) throws Exception {
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java]: I think you should keep List<DlpPositiveItem> here and not define a DlpPositiveItemList

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java [hunk 3]
```diff
@@ -21,8 +29,18 @@ public DlpPositiveItemServiceImpl(DlpPositiveItemDAO dlpPositiveItemDAO) {
...
+        DlpPositiveItemList dlpPositiveItemList = new DlpPositiveItemList();
+        List<DlpPositiveItemEntity> dlpPositiveItemEntities = dlpPositiveItemDAO.findAll();
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java]: Here you should not make a find all. 
Use offset and limit to make a sql query containing a limit option

Comment (rdenarie) [reply commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java]: You can inspire by https://github.com/exoplatform/app-center/blob/develop/app-center-services/src/main/java/org/exoplatform/appcenter/dao/ApplicationDAO.java#L39

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: commons-dlp/pom.xml (hunk 1 above)
Comment (AzmiTouil) [reply commons-dlp/pom.xml]: Removed!

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java (hunk 2 above)
Comment (AzmiTouil) [reply commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java]: Done!

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java (hunk 3 above)
Comment (AzmiTouil) [reply commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java]: Done!

**Review by rdenarie — DISMISSED** [review]

**Review by rdenarie — DISMISSED** [review]

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/dto/DlpPositiveItemList.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
+package org.exoplatform.commons.dlp.dto;
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/dto/DlpPositiveItemList.java]: Should this class be removed ?

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpItemRestServices.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                                        @ApiParam(value = "Limit", required = false, defaultValue = "20") @QueryParam("limit") int limit) {
+
+        if (!userACL.isSuperUser() && !userACL.isUserInGroup(userACL.getAdminGroups())) {
+            throw new WebApplicationException(Response.Status.UNAUTHORIZED);
+        }
+
+        try {
+            List<DlpPositiveItem> dlpPositiveItems = dlpPositiveItemService.getDlpPositivesItems(offset, limit);
+            Long size = dlpPositiveItemService.getDlpPositiveItemsCount();
+            DlpPositiveItemList dlpPositiveItemList = new DlpPositiveItemList();
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpItemRestServices.java]: Why do you need a type DlpPositiveItemList ?
What it bring more than a simple arrayList ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/dto/DlpPositiveItemList.java (hunk 4 above)
Comment (AzmiTouil) [reply commons-dlp/src/main/java/org/exoplatform/commons/dlp/dto/DlpPositiveItemList.java]: Removed!

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpItemRestServices.java (hunk 5 above)
Comment (AzmiTouil) [reply commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpItemRestServices.java]: updated !

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpItemRestServices.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            notes = "Return list of dlp positive items in json format")
+    @ApiResponses(
+            value = {@ApiResponse(code = HTTPStatus.OK, message = "Request fulfilled"),
+                    @ApiResponse(code = HTTPStatus.UNAUTHORIZED, message = "Unauthorized operation"),
+                    @ApiResponse(code = HTTPStatus.INTERNAL_ERROR, message = "Internal server error"),}
+    )
+    public Response getDlpPositiveItems(@ApiParam(value = "Offset", required = false, defaultValue = "0") @QueryParam("offset") int offset,
+                                        @ApiParam(value = "Limit", required = false, defaultValue = "20") @QueryParam("limit") int limit) {
+
+        if (!userACL.isSuperUser() && !userACL.isUserInGroup(userACL.getAdminGroups())) {
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpItemRestServices.java]: Why do you check permissions ? 
We have the @RolesAllowed annotation which check already that point.

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java [hunk 7]
```diff
@@ -44,4 +51,26 @@ public void deleteDlpPositiveItem(Long itemId) {
     public DlpPositiveItemEntity getDlpPositiveItemByReference(String itemReference) {
         return dlpPositiveItemDAO.findDlpPositiveItemByReference(itemReference);
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java]: All function of the service should return a DTOObject (in your case DlpPositiveItem
The object DlpPositiveItemEntity is reserved to be used between the service and the DAO.

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/DlpPositiveItemService.java [hunk 8]
```diff
@@ -1,16 +1,18 @@
...
 public interface DlpPositiveItemService {
 
-    List<DlpPositiveItemEntity> getDlpPositivesItems();
+    List<DlpPositiveItem> getDlpPositivesItems(int offset, int limit) throws Exception;
 
     void addDlpPositiveItem(DlpPositiveItemEntity dlpPositiveItemEntity);
     
     void deleteDlpPositiveItem(Long itemId);
     
     DlpPositiveItemEntity getDlpPositiveItemByReference(String itemReference);
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/DlpPositiveItemService.java]:
```suggestion
    DlpPositiveItem getDlpPositiveItemByReference(String itemReference);
```

File: commons-dlp/src/test/java/org/exoplatform/commons/dlp/service/DlpPositiveItemServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        //Given
+        DlpPositiveItemEntity dlpPositiveItemEntity = new DlpPositiveItemEntity();
+        dlpPositiveItemEntity.setType("file");
+        dlpPositiveItemEntity.setTitle("file");
+        //When
+        dlpPositiveItemService.addDlpPositiveItem(dlpPositiveItemEntity);
+        //Then
+        verify(dlpPositiveItemDAO, times(1)).create(dlpPositiveItemEntity);
+    }
+}
```
Comment (rdenarie) [inline commons-dlp/src/test/java/org/exoplatform/commons/dlp/service/DlpPositiveItemServiceTest.java:99]: Can you add tests for 
getDlpPositivesItems
getDlpPositiveItemByReference

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpItemRestServices.java (hunk 6 above)
Comment (AzmiTouil) [reply commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpItemRestServices.java]: updated !

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java (hunk 7 above)
Comment (AzmiTouil) [reply commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java]: updated !

File: commons-dlp/src/test/java/org/exoplatform/commons/dlp/service/DlpPositiveItemServiceTest.java (hunk 9 above)
Comment (AzmiTouil) [reply commons-dlp/src/test/java/org/exoplatform/commons/dlp/service/DlpPositiveItemServiceTest.java:99]: Done !

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java [hunk 10]
```diff
@@ -41,7 +48,35 @@ public void deleteDlpPositiveItem(Long itemId) {
...
+        return dlpPositiveItems;
+    }
+
+    private DlpPositiveItem fillDlpPositiveItemFromEntity(DlpPositiveItemEntity dlpPositiveItemEntity) throws Exception {
+        DlpPositiveItem dlpPositiveItem = new DlpPositiveItem();
+        dlpPositiveItem.setId(dlpPositiveItemEntity.getId());
+        dlpPositiveItem.setType(dlpPositiveItemEntity.getType());
+        dlpPositiveItem.setKeywords(dlpPositiveItemEntity.getKeywords());
+        dlpPositiveItem.setAuthor(dlpPositiveItemEntity.getAuthor());
+        dlpPositiveItem.setAuthorFullName(organizationService.getUserHandler().findUserByName(dlpPositiveItemEntity.getAuthor()).getDisplayName());
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java]: I prefer that DlpService not depends of orgService
You should return username here, and at display, get userfullname with rest service

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java (hunk 10 above)
Comment (AzmiTouil) [reply commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java]: Updated !

**Review by rdenarie — APPROVED** [review]

**Pull request merged**